apps/sms/js/activity_handler.js

```diff
@@ -250,54 +259,30 @@ var ActivityHandler = {
     });
   },
 
-  // The unsent confirmation dialog provides 2 options: edit and discard
-  // discard: clear the message user typed
-  // edit: continue to edit the unsent message and ignore the activity
-  displayUnsentConfirmation: function ah_displayUnsentConfirmtion(activity) {
-    var msgDiv = document.createElement('div');
-    msgDiv.innerHTML = '<h1 data-l10n-id="unsent-message-title"></h1>' +
-                       '<p data-l10n-id="unsent-message-description"></p>';
-    var options = new OptionMenu({
-      type: 'confirm',
-      section: msgDiv,
-      items: [{
-        l10nId: 'unsent-message-option-edit',
-        method: function editOptionMethod() {
-          // we're already in message app, we don't need to do anything
-        }
-      },
-      {
-        l10nId: 'unsent-message-option-discard',
-        method: (activity) => {
-          ThreadUI.discardDraft();
-          this.launchComposer(activity);
-        },
-        params: [activity]
-      }]
-    });
-    options.show();
-  },
-
-  // Launch the UI properly
-  launchComposer: function ah_launchComposer(activity) {
-    Navigation.toPanel('composer', { activity: activity });
-  },
-
   // Check if we want to go directly to the composer or if we
   // want to keep the previously typed text
   triggerNewMessage: function ah_triggerNewMessage(body, number, contact) {
-     var activity = {
+    var composerArgs = {
+      activity: {
         body: body || null,
         number: number || null,
         contact: contact || null
-      };
+      }
+    };
 
     if (Compose.isEmpty()) {
-      this.launchComposer(activity);
-    } else {
-      // ask user how should we do
-      ActivityHandler.displayUnsentConfirmation(activity);
+      Navigation.toPanel('composer', composerArgs);
+      return;
     }
+
+    Utils.confirm(
```

Comment:
Originally I was thinking that this should be triggered when in new activity, but the fact is opening new activity from other app will not trigger this(maybe it's because of new activity became inline, and it's reasonable from my POV because the message is saved as draft). After diving further I found it was mainly use for ActivityPicker.sendMessage. I'm not sure if it's suitable to reuse Activity.toView in that case especially it's not related to activity, but leaving this part in activity handler and no activity will actually uses it seems weird to me. Please correct me if I misunderstood.

So, here is some idea(but maybe not in this patch): Apply new confrim/toPanel logic in ActivityPicker.sendMessage instead of reuse ActivityHandler.toView(if didn't enter the thread directly, maybe we can refine it as well), and we can revisit if this part is necessary for ActivityHandler, wdyt?